enclave/src/main/java/org/hyperledger/besu/enclave/VertxRequestTransmitter.java

```diff
@@ -72,14 +75,11 @@ public VertxRequestTransmitter(final HttpClient httpClient) {
       }
       return result.get();
     } catch (final ExecutionException e) {
+
       if (e.getCause() instanceof EnclaveException) {
         throw (EnclaveException) e.getCause();
       }
-      throw new EnclaveException("Failure during reception", e);
-    } catch (final InterruptedException e) {
-      throw new EnclaveException("Task interrupted while waiting for Enclave response.", e);
-    } catch (final Exception e) {
-      throw new EnclaveException("Other error", e);
+      throw e;
```

Comment:
I wonder if we are providing any value exposing the `InterruptedException` and `ExecutionException` checked exception here. Maybe we can add something to make `EnclaveExceptions` more granular? Not sure on the best approach forward but forcing the checked exceptions into this interface seems a bit overkill.

Reply:
Agreed - the interrupted/executionException leaks that the comms are async, so probably shouldn't be visible outside of the Enclave.

Reply:
The problem is we're using the EnclaveException as an overused goto statement. i.e we're branching based on exception and then parsing the message of the exception to choose what to do. So what we have, currently, is this brittle connascence between the string in the exception and the behaviour of the api and the private transaction processor.

Specifically EnclaveException It's used for various things including:
- when there's a 404 the private transaction processor should not process the private transaction
- it's also used to magically match up these errors for api calls https://github.com/hyperledger/besu/blob/master/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/JsonRpcError.java#L108
- you can't easily tell the difference outside of the enclave what type of error it is ie 4xx your error 5xx my error (and therefore you can't tell the user).
- all the api responses are 400 rather than respecting http status codes.

Granular exceptions might be the way to go but we will end up with an explosion of types, `BadRequestEnclaveException` `NotFoundEnclaveException` etc that might not be a bad thing as at least you'll be able to navigate the code more easily.

Reply:
Okay I've wrapped everything up in *EnclaveExceptions according to their type of error. I noticed that the http response codes from Orion are not correct. Which is causing some dificulty here, as highlighted by the failing acceptance tests. @lucassaldanha @rain-on let me know what you think